python/pyarrow/tensorflow/build.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -ex
+
+PYARROW_TENSORFLOW_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
```

Comment:
I'm genuinely curious about why this (which at a glance looks like chicanery to me) is required versus handling this at the pyarrow (or libarrow) build stage. Would it be possible to package pyarrow with TF support out of the box?

Reply:
I did it in the pyarrow build stage in https://github.com/apache/arrow/pull/2046 but it was correctly pointed out that the TensorFlow op probably needs to be built on the user's computer to be compatible with all versions of tensorflow. This is trying to make it so we need minimal requirements (only g++ and TensorFlow headers which come with the pip package). I.e. no CMake. I'm very open to suggestions if somebody has a better way of doing this. The goal is to be compatible with as many TensorFlow versions and user configurations as possible (tensorflow-cpu, tensorflow-gpu, different versions etc) and make it robust. This will probably need some experimentation as it will be used on many different setups.